### PR TITLE
Junit test

### LINKED ISSUE #142: TaskList Test Coverage
Reach at least 90% test coverage for TaskList class

### PR DESCRIPTION
**Pull request by JTWang2000**
Add Test for TaskList which also covers Tasks.

### CODE REVIEW

**Comment by rsanchez-macias** [comment]
Nice job with the JUnit tests :) I wanted to merge it but there seems to be a conflict

**Comment by JTWang2000** [comment]
> Nice job with the JUnit tests :) I wanted to merge it but there seems to be a conflict

I am checking which part of style not pass but on my Intellij, my coding standard should be okay.

**Comment by JTWang2000** [comment]
image
It tells me that the changeTask failed but on My Intellij, it works fine.

**Comment by rsanchez-macias** [comment]
That's strange, do you think you could share what gradle says when you build the project locally?

**Comment by rsanchez-macias** [comment]
Oh nvm I am able to see them here 😅

**Comment by rsanchez-macias** [comment]
Did you see that error about the incorrect indentation?

**Comment by JTWang2000** [comment]
> Did you see that error about the incorrect indentation?

But I cannot edit on my own repo now. Strange.

**Comment by JTWang2000** [comment]
> > Did you see that error about the incorrect indentation?
> 
> But I cannot edit on my own repo now. Strange.

Maybe becasue I merge with your branch, then I cannot edit on my own repo?

**Comment by rsanchez-macias** [comment]
Can you check this part?
Screen Shot 2020-04-09 at 9 16 02 PM

Maybe we could merge the PR and then I could help you find the problem?

File: src/main/java/seedu/tasklist/TaskList.java [hunk 1]
```diff
@@ -52,21 +52,8 @@ public void changeLocation(int taskIndex, String newLocation) {
...
-    public void changeCategory(int taskIndex, String newCategory) {
-        String oldCategory = taskList.get(taskIndex).getCategory();
-        this.categoryMap.get(oldCategory).remove(taskIndex);
-        taskList.get(taskIndex).setCategory(newCategory);
-        updateCategoryMap(newCategory, taskIndex);
-    }
-
-
+  
+  //Other methods:
```
Comment (rsanchez-macias) [inline src/main/java/seedu/tasklist/TaskList.java]: The style error seems to be here because of the indentation of the "//Other methods" comment

Comment (JTWang2000) [reply src/main/java/seedu/tasklist/TaskList.java]: image
Interestingly, all the constants cannot be recognized. I am totally confused.

**Comment by JTWang2000** [comment]
I do not really know what happens, but all the loading becomes unavaliable. Interesting.

**Comment by rsanchez-macias** [comment]
I am not sure why that happened, and it is strange that it doesn't recognize the constants

**Comment by rsanchez-macias** [comment]
Have you pulled the latest changes from the team repo?

**Comment by JTWang2000** [comment]
> I am not sure why that happened, and it is strange that it doesn't recognize the constants

So I cannot really edit this branch. Is there anyway that I cannot edit the code on github temporarily?

**Comment by JTWang2000** [comment]
Back to previous problem:

image

The changeTask works fine on my computer but works bad on here.
Also, my IntelliJ is suffering from some unknown issues. So I can only try push here again and again for testing.

File: src/test/java/seedu/tasklist/TaskListTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        Integer output = tasks.getListSize();
+        Integer expected = 2;
+        assertEquals(output,expected);
+        tasks.deleteTask(1);
+        Integer outputAfterDelete = tasks.getListSize();
+        Integer expectedAfterDelete = 1;
+        assertEquals(outputAfterDelete,expectedAfterDelete);
+    }
+
+    @Test
```
Comment (Bencotti) [inline src/test/java/seedu/tasklist/TaskListTest.java]: Error should be here, because this @Test was not removed @JTWang2000

Comment (Bencotti) [reply src/test/java/seedu/tasklist/TaskListTest.java]: I have removed it via Git edit, checking to see if that's the cause now

Comment (Bencotti) [reply src/test/java/seedu/tasklist/TaskListTest.java]: Problem resolved

Comment (JTWang2000) [reply src/test/java/seedu/tasklist/TaskListTest.java]: Thanks for your help! I really appreciate it.

**Comment by JTWang2000** [comment]
The changeTask method is still commented now.

**Pull request merged**